docs/lib/datastore/getting-started.md

```diff
@@ -9,44 +9,161 @@ Amplify DataStore provides a programming model for leveraging shared and distrib
+
+The base structure for a DataStore app is created by adding a new GraphQL API to your app.
+
+```console
+amplify add api
+```
+
+During the API configuration process select **GraphQL** as the API type and reply to the questions as follows. Make sure you respond **Yes, I want to make some additional changes** when prompted for **advanced settings** and turn on **conflict detection**. This setting is critical when syncing data to the cloud since the conflict resolution strategy is what allows local data to be reconciled with data from the cloud backend.
+
+```console
+? Please select from one of the below mentioned services:
+    `GraphQL`
+? Provide API name:
+    `BlogAppApi`
+? Choose the default authorization type for the API
+    `API key`
+? Enter a description for the API key:
+    `BlogAPIKey`
+? After how many days from now the API key should expire (1-365):
+    `365`
+? Do you want to configure advanced settings for the GraphQL API
+    `Yes, I want to make some additional changes.`
+? Configure additional auth types?
+    `No`
+? Configure conflict detection?
+    `Yes`
+? Select the default resolution strategy
+    `Auto Merge`
+? Do you have an annotated GraphQL schema?
+    `No`
+? Do you want a guided schema creation?
+    `No`
+? Provide a custom type name
+    `Post`
+```
+
+<amplify-callout warning>
+
+**Troubleshooting:** without the **conflict detection** configuration cloud sync will fail. In that case use `amplify update api` and choose **Enable DataStore for entire API** (this option will enable the conflict detection as described above).
+
+</amplify-callout>
+
+## Idiomatic persistence
+
+DataStore relies on platform standard data structures to represent the data schema in an idiomatic way. The persistence language is composed by data types that satisfies the `Model` interface and operations defined by common verbs such as `save`, `query` and `delete`.
+
+### Data schema
+
+The first step to create an app backed by a persistent datastore is to **define a schema**. DataStore uses GraphQL schema files as the definition of the application data model. The schema contains data types and relationships that represent the app's functionality.
 
 ### Sample schema
 
-For the purpose of this guide, let's use the following simple GraphQL schema:
+For the next steps, let's start with a schema for a small blog application. It has a single model, a `Post`. New types and constructs will be added to this base schema as more concepts are presented.
 
-```graphql
-enum PostStatus {
-  ACTIVE
-  INACTIVE
-}
+Open the `schema.graphql` file located by default at `amplify/backend/{api_name}/` and **define the model** `Post` as follows.
 
+```graphql
 type Post @model {
   id: ID!
   title: String!
-  rating: Int!
   status: PostStatus!
+  rating: Int
+  content: String
+}
+
+enum PostStatus {
```

Comment:
Is the enum necessary for a getting started or should be make this as basic as possible?

Reply:
@dabit3 that a very good callout, but tbh as much as I want to make it as simple as possible, right now we lack a more organized page on how to define schemas and all the capabilities for newcomers of DataStore. We do have a "transformer" documentation outside of the DataStore scope but that not only takes the developer to another page but also into a much deeper GraphQL context than what DataStore is trying to expose.

But I would say we should take that as a goal on the next improvement we need to add to DataStore docs: a schema mapping page, tools like Hibernate and Rails ActiveRecord invest heavily into document best practices and different scenarios when defining a data schema (your recipes repo could serve as a base I think!).